### PR TITLE
Match kwargs in SubgrpahMatcher

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Pattern node and target node must have identical kwargs now... 

Use envvar `LOGLEVEL=INFO` to turn on the logging message for easier debugging... 

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85617

### CODE REVIEW

File: torch/fx/passes/utils/matcher_utils.py [hunk 1]
```diff
@@ -172,7 +179,14 @@ def _match_nodes(self, pn: Node, gn: Node, match: InternalMatch) -> bool:
         pn_flatten_args, _ = pytree.tree_flatten(pn.args)
         gn_flatten_args, _ = pytree.tree_flatten(gn.args)
 
-        if len(pn_flatten_args) == len(gn_flatten_args):
+        if pn.kwargs.keys() == gn.kwargs.keys():
```
Comment (jerryzh168) [inline torch/fx/passes/utils/matcher_utils.py:182]: nit: do we need to make sure the keys are the same as well?

Comment (SherlockNoMad) [reply torch/fx/passes/utils/matcher_utils.py:182]: pn.kwargs.keys() == gn.kwargs.keys() 

hmm, is something missing with this check?

File: test/fx/test_subgraph_rewriter.py [hunk 2]
```diff
@@ -656,3 +668,99 @@ def replacement(x):
...
+
+        found_repalcement_node = False
+        for node in traced.graph.nodes:
+            if node.target == wrapped_gemm_bias_mul:
+                found_repalcement_node = True
+                break
+
+        self.assertTrue(found_repalcement_node)
+
+    def test_subgraph_rewriter_local_revert(self):
```
Comment (jerryzh168) [inline test/fx/test_subgraph_rewriter.py:708]: could you add some description on how this tests the functionality?

**Review by jerryzh168 — APPROVED** [review]

**Comment by jerryzh168** [comment]
looks good, might be good to separate the functionality change and logging change in the future

File: test/fx/test_subgraph_rewriter.py (hunk 2 above)
Comment (SherlockNoMad) [reply test/fx/test_subgraph_rewriter.py:708]: Done.

**Comment by SherlockNoMad** [comment]
> looks good, might be good to separate the functionality change and logging change in the future

Thanks for reviewing this!! Will try to keep it as separated PR in the future.

**Review by davidberard98 — APPROVED** [review]
Duplicate Jerry's stamp

**Pull request closed without merging**